build_tools/benchmarks/common/benchmark_definition.py

```diff
@@ -443,3 +443,113 @@ def from_json_str(json_str: str):
         BenchmarkRun.from_json_object(b) for b in json_object["benchmarks"]
     ]
     return results
+
+
+@dataclass(frozen=True)
+class CompilationInfo(object):
+  model_name: str
+  model_tags: Sequence[str]
+  model_source: str
+  target_arch: str
+  bench_mode: Sequence[str]
+
+  def __str__(self):
+    if self.model_tags:
+      tags = ",".join(self.model_tags)
+      model_part = f"{self.model_name} [{tags}] ({self.model_source})"
+    else:
+      model_part = f"{self.model_name} ({self.model_source})"
+    bench_mode_str = ",".join(self.bench_mode)
+    return f"{model_part} {self.target_arch} {bench_mode_str}"
+
+  def to_json_object(self) -> Dict[str, Any]:
+    return {
+        "model_name": self.model_name,
+        "model_tags": self.model_tags,
+        "model_source": self.model_source,
+        "target_arch": self.target_arch,
+        "bench_mode": self.bench_mode,
+    }
```

Comment:
I think you can use asdict, which will also work for recursive dataclasses. I think ultimately the right way to do this is to use a custom json encoder/decoder rather than having extra methods on each dataclass


```suggestion
    return self.asdict()
```

Again, with this replacement, it's not clear that the extra method is really worth it.

Reply:
`asdict()` can recursively serialize the dataclass. So I removed all `to_json_object()` methods and just let caller to serialize it.